### PR TITLE
updates actions with source output, fixes bug

### PR DESCRIPTION
**Pull request by pivotal-david-osullivan**
## Summary
Adds source/source-sha outputs to the following actions:

* Github Release - this required grabbing the original TagName from the Release object, since the action overwrites the TagName as it loops through releases.
* Gradle
* IBM Semeru
* Leiningen
* Liberty - this adds sources only if the dependency contains 'openliberty', since the 'websphere' deps are not OSS

Other action updates contain a bug fix when checking the list of sources for empty, it previously checked for `nil` which won't happen on an empty map, it now checks for length >0

The Skywalking action fixes a bug where an extra `/` was added to the dependency/binary URI

## Checklist

* [ ] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: actions/github-release-dependency/main.go [hunk 1]
```diff
@@ -122,14 +125,27 @@ func main() {
 		for _, a := range r.Assets {
 			if g.MatchString(*a.Name) {
 				versions[*r.TagName] = *a.BrowserDownloadURL
+				sources[*r.TagName] = fmt.Sprintf("https://github.com/%s/%s/archive/refs/tags/%s.tar.gz", o, repo, originalTagName[*r.TagName])
```
Comment (anthonydahanne) [inline actions/github-release-dependency/main.go:128]: how is `originalTagName[*r.TagName]` different from `*r.TagName` ?

File: actions/gradle-dependency/main.go [hunk 2]
```diff
@@ -45,13 +45,25 @@ func main() {
...
+	latestSource := actions.Outputs{}
+	if len(sources) != 0{
+		latestSource["source"] = sources[latestVersion.Original()]
+	}
+
+	o, err := actions.NewOutputs(versions[latestVersion.Original()], latestVersion,  latestSource)
+	if err != nil {
 		panic(err)
-	} else {
+	}else {
```
Comment (anthonydahanne) [inline actions/gradle-dependency/main.go:66]: in this codebase,  there's usually a space there

File: actions/leiningen-dependency/main.go [hunk 3]
```diff
@@ -55,9 +57,19 @@ func main() {
...
+	}
+	latestSource := actions.Outputs{}
+	if len(sources) != 0{
+		latestSource["source"] = sources[latestVersion.Original()]
+	}
+
+	o, err := actions.NewOutputs(versions[latestVersion.Original()], latestVersion,  latestSource)
+	if err != nil {
+		panic(err)
+	}else {
```
Comment (anthonydahanne) [inline actions/leiningen-dependency/main.go:72]: keep the space

File: actions/npm-dependency/main.go [hunk 4]
```diff
@@ -58,15 +58,27 @@ func main() {
...
+	latestSource := actions.Outputs{}
+	if len(sources) != 0{
+		latestSource["source"] = sources[latestVersion.Original()]
+	}
+
+	o, err := actions.NewOutputs(versions[latestVersion.Original()], latestVersion,  latestSource)
+	if err != nil {
 		panic(err)
-	} else {
+	}else {
```
Comment (anthonydahanne) [inline actions/npm-dependency/main.go:81]: keep the space

File: actions/npm-dependency/main.go [hunk 5]
```diff
@@ -58,15 +58,27 @@ func main() {
 	}
 
 	versions := make(actions.Versions)
+	sources := make(map[string]string)
 	for k, v := range raw.Versions {
 		if p := versionPattern.MatchString(k); p {
 			versions[k] = v.Dist.Tarball
+			sources[k] = v.Dist.Tarball
```
Comment (anthonydahanne) [inline actions/npm-dependency/main.go:65]: looks good!

File: actions/skywalking-dependency/test [hunk 6]
```diff
@@ -0,0 +1,15 @@
+sha256<<20d8a77d04df9ce84bc2f94976ac980b
```
Comment (anthonydahanne) [inline actions/skywalking-dependency/test:1]: how is this file used?

**Review by anthonydahanne — CHANGES_REQUESTED** [review]

File: actions/skywalking-dependency/test (hunk 6 above)
Comment (pivotal-david-osullivan) [reply actions/skywalking-dependency/test]: Accidental commit!

File: actions/github-release-dependency/main.go (hunk 1 above)
Comment (pivotal-david-osullivan) [reply actions/github-release-dependency/main.go:128]: I mentioned in the comment but this action overwrites the tag name with the normalised version, so I have to grab the original before this happens in order to build the source url using the proper tag name

File: actions/gradle-dependency/main.go (hunk 2 above)
Comment (pivotal-david-osullivan) [reply actions/gradle-dependency/main.go:66]: Could you add a suggestion with these spaces and I can commit them? 🙏

Comment (anthonydahanne) [inline actions/gradle-dependency/main.go:66]:
```suggestion
	} else {
```

File: actions/leiningen-dependency/main.go (hunk 3 above)
Comment (anthonydahanne) [inline actions/leiningen-dependency/main.go:72]:
```suggestion
	} else {
```

File: actions/npm-dependency/main.go (hunk 4 above)
Comment (anthonydahanne) [inline actions/npm-dependency/main.go:81]:
```suggestion
	} else {
```

**Review by anthonydahanne — APPROVED** [review]

**Pull request merged**